torchq/mesolve.py

```diff
@@ -0,0 +1,65 @@
+from typing import Any, Dict, List, Optional
+
+import numpy as np
+import torch
+import torchdiffeq
+
+
+def mesolve(
```

Comment:
Pour les fonctions principales comme celle-ci, je trouve plus lisible de mettre le typing dans le docstring. Surtout si on commence à avoir des `Union[QTensor, list of QTensor, nn.Module]` ou truc comme ça pour `H`.
Pour toutes les autres fonctions par contre, le typing dans les arguments c'est très bien!

Reply:
Oui c'est vrai. Une autre option, c'est de créer des types agrégés :
```python
from typing import Union
TimeQTensor = Union[QTensor, List[QTensor], ...]

def mesolve(
    H: TimeQTensor,
# ....
```

Reply:
Ho and we should speak English, in case an external collaborator joins or for our future curious users digging the old PRs 😋

Reply:
As you wish, I don't have a super strong opinion on the subject. But I know that when I was reading the source code of torchdyn I was having a hard time understanding the typing because I was not familiar with it. While torchdiffeq is not using any typing, and I found it easier to understand (if variable names are well chosen). I'm guessing most python users are not familiar with typing either, so only keeping it in the dosctring would be probably make the function more accessible.